### PR TITLE
chore: update data explorer scaler to fix deprecation

### PR DESCRIPTION
**Pull request by JorTurFer**
Signed-off-by: Jorge Turrado <jorge_turrado@hotmail.es>

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes https://github.com/kedacore/keda/issues/2989

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -54,6 +54,7 @@ To learn more about our roadmap, we recommend reading [this document](ROADMAP.md
 - **ActiveMQ Scaler:** Add CorsHeader information to ActiveMQ Scaler ([#2884](https://github.com/kedacore/keda/issues/2884))
 - **Azure Application Insights Scaler:** Provide support for non-public clouds ([#2735](https://github.com/kedacore/keda/issues/2735))
 - **Azure Blob Storage Scaler:** Add optional parameters for counting blobs recursively ([#1789](https://github.com/kedacore/keda/issues/1789))
+- **Azure Data Explorer:** Replace deprecated function `iter.Next()` in favour of `iter.NextRowOrError()` ([#2989](https://github.com/kedacore/keda/issues/2989))
```
Comment (zroubalik) [inline CHANGELOG.md:57]: I would put it into other section as it is just internal change

**Review by zroubalik — COMMENTED** [review]
LGTM, just fix the changelog

File: CHANGELOG.md (hunk 1 above)
Comment (JorTurFer) [reply CHANGELOG.md]: moved to `### Other` section

**Review by zroubalik — APPROVED** [review]

**Pull request merged**